PythonVisualizations/SimpleQueue.py

```diff
@@ -6,357 +6,238 @@
         n = self.list[self.front]
         self.list[self.front] = None
         
         #increment front
         self.front += 1
-        
+        # delete the associated display objects
+        self.canvas.delete(n.display_shape)
+        self.canvas.delete(n.display_val)
         #deal with wraparound
         if self.front == self.size:
             self.front = 0
-            
-        self.nItems -= 1
-        
-        # delete the associated display objects
-        canvas.delete(n.display_shape)
-        canvas.delete(n.display_val)
-        
-        self.onOffButtons()
-        
-        # update window
-        window.update()
-    
-    def removeRear(self):
-        
-        #if the queue is empty, exit
-        if self.size == 0:
-            return None
-        
-        #get the value at rear, and then set it to None so it will be garbage collected
-        n = self.list[self.rear]
-        self.list[self.rear] = None
-        
-        #decrement rear
-        self.rear -= 1
-        
-        #deal with wraparound
-        if self.rear == -1:
-            self.rear = self.size-1
-            
+            self.moveItemsBy(self.frontArrow, (self.CELL_SIZE * self.size * -1, 0))
+        #move arrow
+        self.moveItemsBy(self.frontArrow, (self.CELL_SIZE, 0))
+        #decrement number of arrows
         self.nItems -= 1
-        
-        # delete the associated display objects
-        canvas.delete(n.display_shape)
-        canvas.delete(n.display_val)
-        
-        self.onOffButtons()
-        
+
         # update window
-        window.update()
-    
+        self.window.update()
+        self.cleanUp(callEnviron)
+
     def display(self):
-        canvas.delete("all")
-        xpos = ARRAY_X0
-        ypos = ARRAY_Y0
 
-        for i in range(self.size):
-            canvas.create_rectangle(xpos + CELL_SIZE*i, ypos, xpos + CELL_SIZE*(i+1), ypos + CELL_SIZE, fill='white', outline='black')
+        self.canvas.delete("all")
+
+        self.makeButtons()
+        self.onOffButtons()
+        xpos = self.ARRAY_X0
+        ypos = self.ARRAY_Y0
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
 
         # go through each Element in the list
         for n in self.list:
-            
+
             # Only loop through the existing elements
             if n:
-                print(n)
+
                 # create display objects for the associated Elements
-                cell = canvas.create_rectangle(xpos, ypos, xpos+CELL_SIZE, ypos+CELL_SIZE, fill=n[1], outline='')
-                cell_val = canvas.create_text(xpos+(CELL_SIZE/2), ypos+(CELL_SIZE/2), text=n[0], font=('Helvetica', '20'))
-    
+                cell = self.canvas.create_rectangle(xpos, ypos, xpos+self.CELL_SIZE, ypos+self.CELL_SIZE, fill=n[1], outline='')
+                cell_val = self.canvas.create_text(xpos+(self.CELL_SIZE/2), ypos+(self.CELL_SIZE/2), text=n[0], font=('Helvetica', '20'))
+
```

Comment:
The `cell` and `cell_val` items should be positioned using the `cellCoords()` and `cellCenter()` functions.

Reply:
would the correct way to call the function be cellcoords(n)?
Also i think this might be left over from other functions where an array is generated initially, but in simplequeue this doesnt happen so i dont think this does anything.

Reply:
Yes the call to `create_rectangle()` should call `cellCoords()` and the call to `create_text()` should call `cellCenter()` to get the coordinates of their respective shapes.  The argument to `cellCoords()` wouldn't be `n`; it would the index of `n` in `self.list`.
I believe you are right that this code only gets invoked if there are items in `self.list` when `display()` is called.  That currently does not happen, since it is only called by `__init__()` and `new()` which both set `self.nItems` to 0.  So you could take this code out.